django_mongodb_backend/indexes.py

```diff
@@ -101,6 +105,180 @@ def where_node_idx(self, compiler, connection):
     return mql
 
 
+class SearchIndex(Index):
+    suffix = "six"
+    _error_id_prefix = "django_mongodb_backend.indexes.SearchIndex"
+
+    def check(self, model, connection):
+        errors = []
+        if not connection.features.supports_atlas_search:
+            errors.append(
+                Warning(
+                    f"This MongoDB server does not support {self.__class__.__name__}.",
+                    hint=(
+                        "The index won't be created. Use an Atlas-enabled version of MongoDB, "
+                        "or silence this warning if you don't care about it."
+                    ),
+                    obj=model,
+                    id=f"{self._error_id_prefix}.W001",
+                )
+            )
+        return errors
+
+    def search_index_data_types(self, field, db_type):
+        """
+        Map a model field's internal type to search index type.
+        Reference: https://www.mongodb.com/docs/atlas/atlas-search/define-field-mappings/#data-types
+        """
+        if field.get_internal_type() == "UUIDField":
+            return "uuid"
```

Comment:
As currently implemented, I believe this should be string: https://github.com/mongodb/django-mongodb-backend/blob/43f477cc2e1a0981e53bc2a60ee0e049939c716f/django_mongodb_backend/base.py#L62

Separately, maybe it's worth confirming that `UUIDField` can't store it values as BSON uuid. Setting [`DatabaseFeatures.has_native_uuid_field = True`](https://github.com/django/django/blob/main/django/db/models/fields/__init__.py#L2753-L2761) raises the error: ValueError: cannot encode native uuid.UUID with UuidRepresentation.UNSPECIFIED. UUIDs can be manually converted to bson.Binary instances using bson.Binary.from_uuid() or a different UuidRepresentation can be configured. See the documentation for UuidRepresentation for more information.

I don't think I tried any further to make that work.